### PR TITLE
[improve][broker][PIP-149]make deletePersistence method async in Namespaces

### PR DESCRIPTION
**Pull request by Pomelongan**
### Motivation

See https://github.com/apache/pulsar/issues/14365

### Verifying this change

Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 
- [x] `doc-not-needed`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 1]
```diff
@@ -1559,19 +1560,37 @@ protected void internalSetPersistence(PersistencePolicies persistence) {
 
     private void doUpdatePersistence(PersistencePolicies persistence) {
         try {
-            updatePolicies(namespaceName, policies -> {
-                policies.persistence = persistence;
-                return policies;
-            });
-            log.info("[{}] Successfully updated persistence configuration: namespace={}, map={}", clientAppId(),
-                    namespaceName, jsonMapper().writeValueAsString(persistence));
+            doUpdatePersistenceAsyn(persistence).get(namespaceResources().getOperationTimeoutSec(), TimeUnit.SECONDS);
```
Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1563]: I thinks the modification of this method is unrelated to this PR,  we'd better do not change it now

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 2]
```diff
@@ -1559,19 +1560,37 @@ protected void internalSetPersistence(PersistencePolicies persistence) {
...
+                throw new RestException(cause);
+            } else {
+                throw (RestException) cause;
+            }
         }
     }
 
+    private CompletableFuture<Void> doUpdatePersistenceAsyn(PersistencePolicies persistence) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        updatePoliciesAsync(namespaceName, policies -> {
```
Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1576]:
```suggestion
        return updatePoliciesAsync(namespaceName, policies -> {
```
It's better return directly I think :)

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 3]
```diff
@@ -1543,10 +1544,10 @@ protected void internalSetRetention(RetentionPolicies retention) {
         }
     }
 
-    protected void internalDeletePersistence() {
-        validateNamespacePolicyOperation(namespaceName, PolicyName.PERSISTENCE, PolicyOperation.WRITE);
-        validatePoliciesReadOnlyAccess();
-        doUpdatePersistence(null);
+    protected CompletableFuture<Void> internalDeletePersistenceAsyn() {
```
[the remark this answers is not in the document]
Comment (Pomelongan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: Thank you for your review. I have revised it

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 4]
```diff
@@ -1559,19 +1560,37 @@ protected void internalSetPersistence(PersistencePolicies persistence) {
...
+    private CompletableFuture<Void> doUpdatePersistenceAsyn(PersistencePolicies persistence) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        updatePoliciesAsync(namespaceName, policies -> {
+            policies.persistence = persistence;
+            return policies;
+        }).thenAccept(v -> {
+            try {
+                log.info("[{}] Successfully updated persistence configuration: namespace={}, map={}", clientAppId(),
+                        namespaceName, jsonMapper().writeValueAsString(persistence));
+            } catch (JsonProcessingException ignore) {}
```
[the remark this answers is not in the document]
Comment (Pomelongan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: Thank you for your review. I have revised it

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java (hunk 2 above)
Comment (Pomelongan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: Thank you for your review. I have revised it

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 5]
```diff
@@ -1572,6 +1572,19 @@ private void doUpdatePersistence(PersistencePolicies persistence) {
...
+    private CompletableFuture<Void> doUpdatePersistenceAsync(PersistencePolicies persistence) {
+        return updatePoliciesAsync(namespaceName, policies -> {
+            policies.persistence = persistence;
+            return policies;
+        }).thenAccept(__ -> log.info("[{}] Successfully updated persistence configuration: namespace={}, map={}",
+                clientAppId(), namespaceName, persistence)
+        ).exceptionally(ex -> {
+            log.error("[{}] Failed to update persistence configuration for namespace {}", clientAppId(), namespaceName,
+                    ex);
+            throw new RestException(ex);
```
Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1584]:
```suggestion
           Throwable cause = FutureUtil.unwrapCompletionException(ex);
            log.error("[{}] Failed to update persistence configuration for namespace {}", clientAppId(), namespaceName,
                    cause);
            throw new RestException(cause);
```

Comment (nodece) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1584]: You can move log logic to v2.

Comment (Pomelongan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: > You can move log logic to v2.

PTAL

Comment (Pomelongan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: PTAL @nodece

**Review by AnonHxy — APPROVED** [review]

**Comment by HQebupt** [comment]
@Pomelongan Please fix CI failure.

**Review by HQebupt — APPROVED** [review]
